### ISSUE TITLE
deeplabv3p backbone 是resnet50 , cpu和gpu 跑出来 模型效果（精度）不一样 ，这是什么原因？

### ISSUE THREAD

**Issue by mingo-doer** [PaddlePaddle/PaddleSeg#1311]
背景：卡证的分割
GPU效果：
![image](https://user-images.githubusercontent.com/28819717/131618735-dcff0370-0666-4ebd-ac7d-9504471cf2cb.png)
CPU 效果：
![image](https://user-images.githubusercontent.com/28819717/131618747-ccc73699-1b19-446c-9f54-6ef85e1f51c0.png)
可以解释一下这是为什么吗

**Comment by michaelowenliu** [comment]
@mingo-doer Hi, could you help us reproduce it? e.g. the trained-weights, and a few instructions. Thanks.

**Comment by mingo-doer** [comment]
> @mingo-doer Hi, could you help us reproduce it? e.g. the trained-weights, and a few instructions. Thanks

Hi, I used  the config file 'deeplabv3p_resnet50_os8_voc12aug_512x512_40k.yml' to train the model.

I can send  trained model via email to reproduce this problem, if you need.

**Comment by mingo-doer** [comment]
_base_: '../_base_/pascal_voc12aug.yml'

batch_size: 6
iters: 80000

train_dataset:
  type: OpticDiscSeg
  dataset_root: 
  transforms:
    - type: Resize
      target_size: [512, 512]
    - type: RandomHorizontalFlip
    - type: Normalize
  mode: train

val_dataset:
  type: OpticDiscSeg
  dataset_root: 
  transforms:
    - type: Resize
      target_size: [512, 512]
    - type: Normalize
  mode: val

optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 4.0e-5

lr_scheduler:
  type: PolynomialDecay
  learning_rate: 0.01
  end_lr: 0
  power: 0.9

loss:
  types:
    - type: CrossEntropyLoss
  coef: [1]

model:
  type: DeepLabV3P
  backbone:
    type: ResNet50_vd
    output_stride: 8
    multi_grid: [1, 2, 4]
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/resnet50_vd_ssld_v2.tar.gz
  backbone_indices: [0, 3]
  aspp_ratios: [1, 12, 24, 36]
  aspp_out_channels: 256
  align_corners: False
  pretrained: null

**Comment by mingo-doer** [comment]
@michaelowenliu could you give us some suggestions？

**Comment by michaelowenliu** [comment]
> > @mingo-doer Hi, could you help us reproduce it? e.g. the trained-weights, and a few instructions. Thanks
> 
> Hi, I used the config file 'deeplabv3p_resnet50_os8_voc12aug_512x512_40k.yml' to train the model.
> 
> I can send trained model via email to reproduce this problem, if you need.

Hi, can you share the link?

**Comment by mingo-doer** [comment]
@michaelowenliu   https://drive.google.com/file/d/1grNhsuGwUVSJPo6MFQSbEvyT1HF6K2U0/view?usp=sharing

**Comment by michaelowenliu** [comment]
> @michaelowenliu https://drive.google.com/file/d/1grNhsuGwUVSJPo6MFQSbEvyT1HF6K2U0/view?usp=sharing

thanks, can you provide the testing image?

**Comment by mingo-doer** [comment]
@michaelowenliu 
sorry, we find it not caused by the difference between cpu and gpu 
The inconsistent accuracy may be caused by  the inference code ‘deploy/python/infer.py ’and the code ‘predict.py’
test image link ：https://drive.google.com/file/d/1RRtye51DamUetz7CmZCu7dbrOpkfvVMg/view?usp=sharing
predict.py：
![image](https://user-images.githubusercontent.com/28819717/131965737-5670087c-8e55-4b13-9083-08dad4275b0e.png)

/infer.py ：
![image](https://user-images.githubusercontent.com/28819717/131965782-1c4c124d-0eee-4713-955a-6ced40ae821c.png)

**Comment by michaelowenliu** [comment]
@mingo-doer Hi, did you follow the "export" steps before running `infer.py`? If not, please refer to [this link](https://github.com/PaddlePaddle/PaddleSeg/blob/release/2.2/docs/export/export/model_export.md).
